File: kyonggiyo-infrastructure/src/main/java/kyonggiyo/persistence/user/UserPersistenceAdapter.java
package kyonggiyo.persistence.user;

import kyonggiyo.application.port.out.user.ExistUserPort;
import kyonggiyo.application.port.out.user.LoadUserPort;
import kyonggiyo.application.port.out.user.SaveUserPort;
import kyonggiyo.domain.user.User;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Component;

@Component
@RequiredArgsConstructor
public class UserPersistenceAdapter implements LoadUserPort, ExistUserPort, SaveUserPort {

    private final UserRepository repository;

    @Override
    public User getById(Long id) {
        return repository.getById(id);
    }

    @Override
    public boolean existByNickname(String nickname) {
        return repository.existByNickname(nickname);
    }

    @Override
    public User save(User user) {
        return repository.save(user);
    }

}

File: kyonggiyo-infrastructure/src/main/java/kyonggiyo/persistence/user/UserPersistenceAdapter.kt
package kyonggiyo.persistence.user

import kyonggiyo.application.user.port.outbound.ExistUserPort
import kyonggiyo.application.user.port.outbound.LoadUserPort
import kyonggiyo.application.user.port.outbound.SaveUserPort
import kyonggiyo.application.user.domain.entity.User
import org.springframework.stereotype.Component

@Component
class UserPersistenceAdapter(
    private val repository: UserRepository,
) : LoadUserPort, ExistUserPort, SaveUserPort {

    override fun existByNickname(nickname: String): Boolean = repository.existByNickname(nickname)

    override fun getById(id: Long): User = repository.getById(id)

    override fun save(user: User): User = repository.save(user)

}